WAVLTree/src/WAVLTree.java

```diff
@@ -3,11 +3,24 @@
  * WAVLTree
  *
  * An implementation of a WAVL Tree.
- * (Haupler, Sen & Tarajan �15)
+ * (Haupler, Sen & Tarajan �15)
  *
  */
 
 public class WAVLTree {
+    WAVLNode root;
+    int size;
+    int height;
+
+    public WAVLTree() {
+        this.root = null;
+    }
+
+    public WAVLTree(WAVLNode root){
+        this.root = root;
+        this.size = 1;
```

Comment:
size might not equal 1 (for example if the root is already the base of an existing subtree). likewise with height

Reply:
Doesn't the size key conflict with the size() method?

Reply:
I was not sure, so I checked. Should not be a problem
https://stackoverflow.com/questions/9960560/java-instance-variable-and-method-having-same-name